packages/flutter/lib/src/rendering/sliver_list.dart

```diff
@@ -118,28 +120,35 @@ class RenderSliverList extends RenderSliverMultiBoxAdaptor {
         }
       }
 
-      final double firstChildScrollOffset = earliestScrollOffset - paintExtentOf(firstChild);
+      double firstChildScrollOffset = earliestScrollOffset - paintExtentOf(firstChild);
       if (firstChildScrollOffset < 0.0) {
```

Comment:
Could we just check if firstChildScrollOffset is smaller than a really small epsilon here instead of doing all the rounding math?

Reply:
The reason I use the rounding is firstChildScrollOffset will later be used to update layoutOffset in line 155. Although I don't think that will make a difference, this will less likely to cumulate precision error.

Reply:
nvm, I just saw we have already defined epsilon to deal with this issue before. Going to reuse that